modules/nw-metallb-frr-k8s-configuration-crd.adoc

```diff
@@ -0,0 +1,221 @@
+    - asn: 64512
+      neighbors:
+      - address: 172.30.0.3
+        asn: 4200000000
+        ebgpMultiHop: true
+        port: 180
+      - address: 172.18.0.6
+        asn: 4200000000
+        port: 179
+----
+
+=== Advertising a prefix to a neighbor
+
+By default, `MetalLB`  does not advertise the prefixes configured as part of a router configuration. 
+In order to advertise them,  you use the `toAdvertise` field.
+
+You can advertise a subset of the prefixes, as in this example:
+
+.Example FRRConfiguration CR
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.30.0.3
+        asn: 4200000000
+        ebgpMultiHop: true
+        port: 180
+        toAdvertise:
+          allowed:
+            prefixes:
+            - 192.168.2.0/24
+      prefixes:
+        - 192.168.2.0/24
+        - 192.169.2.0/24
+----
+
+Alternatively, you can advertise all of the prefixes, as in this example:
+
+.Example FRRConfiguration CR
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.30.0.3
+        asn: 4200000000
+        ebgpMultiHop: true
+        port: 180
+        toAdvertise:
+          allowed:
+            mode: all
+      prefixes:
+        - 192.168.2.0/24
+        - 192.169.2.0/24
+----
+
+=== Receiving prefixes from a neighbor
+
+By default, `MetalLB` does not processe any prefixes advertised by a neighbor. You can use the `toReceive` field  to process such addresses.
+
+You can configure for a subset of the prefixes, as in this example:
+
+.Example FRRConfiguration CR
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+          asn: 64512
+          port: 179
+          toReceive:
+            allowed:
+              prefixes:
+              - prefix: 192.168.1.0/24
+              - prefix: 192.169.2.0/24
+                ge: 25 <1>
+                le: 28 <1>
+----
+<1> The prefix is applied if the prefix length is less than or equal to the `le` prefix length and greater than or equal to the `ge` prefix length.
+
+Alternatively, you can configure FRR to handle all the prefixes announced, as in this example:
+
+.Example FRRConfiguration CR
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+          asn: 64512
+          port: 179
+          toReceive:
+            allowed:
+              mode: all
+----
+
+== Adding a raw configuration
```

Comment:
I'd change this to: the api allows to add a raw configuration, but this is not supported by RH (this is because we don't want to support any possible configuration available).

Probably also worth removing the example to discourage the users.
We must highlight that this field can't be used in ocp

Reply:
@fedepaol would it be best just to remove this altogether for now? Otherwise I'm not sure there's much value in mentioning `rawConfig` and saying it can't be used in OCP.

Reply:
done